src/code/z_play.c

```diff
@@ -1,81 +1,1545 @@
+                Camera_GetQuakeOffset(&sp78, GET_ACTIVE_CAM(this));
+                SkyboxDraw_Draw(&this->skyboxCtx, gfxCtx, this->skyboxId, 0, this->view.eye.x + sp78.x,
+                                this->view.eye.y + sp78.y, this->view.eye.z + sp78.z);
+            }
+
+            // envCtx.precipitation[PRECIP_RAIN_CUR]
+            if (this->envCtx.unk_F2[1] != 0) {
+                Environment_DrawRain(this, &this->view, gfxCtx);
+            }
+        }
+
+        Environment_FillScreen(gfxCtx, 0, 0, 0, this->bgCoverAlpha, 1);
+
+        Actor_DrawAll(this, &this->actorCtx);
+
+        if (!this->envCtx.sunMoonDisabled) {
+            sp25C.x = this->view.eye.x + this->envCtx.unk_4;
+            sp25C.y = this->view.eye.y + this->envCtx.unk_8;
+            sp25C.z = this->view.eye.z + this->envCtx.unk_C;
+            Environment_DrawSunLensFlare(this, &this->envCtx, &this->view, gfxCtx, sp25C);
+        }
+
+        Environment_DrawCustomLensFlare(this);
+
+        if (R_PLAY_FILL_SCREEN_ON) {
+            Environment_FillScreen(gfxCtx, R_PLAY_FILL_SCREEN_R, R_PLAY_FILL_SCREEN_G, R_PLAY_FILL_SCREEN_B,
+                                   R_PLAY_FILL_SCREEN_ALPHA, 3);
+        }
+
+        switch (this->envCtx.fillScreen) {
+            case 1:
+                Environment_FillScreen(gfxCtx, this->envCtx.screenFillColor[0], this->envCtx.screenFillColor[1],
+                                       this->envCtx.screenFillColor[2], this->envCtx.screenFillColor[3], 3);
+                break;
+            default:
+                break;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80167DE4.s")
+        if (this->envCtx.sandstormState != 0) {
+            Environment_DrawSandstorm(this, this->envCtx.sandstormState);
+        }
+
+        if (this->unk_18876 != 0) {
+            Environment_FillScreen(gfxCtx, 0, 0, 0, this->unk_18876, 3);
+        }
+
+        DebugDisplay_DrawObjects(this);
+        Play_DrawMotionBlur(this);
+
+        if (((R_PAUSE_MENU_MODE == 1) || (gTrnsnUnkState == 1)) || (R_PICTOGRAPH_PHOTO_STATE == 1)) {
+            Gfx* sp74;
+            Gfx* sp70 = POLY_OPA_DISP;
+
+            sp74 = Graph_GfxPlusOne(sp70);
+            gSPDisplayList(OVERLAY_DISP++, sp74);
+            this->pauseBgPreRender.fbuf = gfxCtx->curFrameBuffer;
+
+            if (R_PAUSE_MENU_MODE == 1) {
+                R_PAUSE_MENU_MODE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else if (R_PICTOGRAPH_PHOTO_STATE == 1) {
+                R_PICTOGRAPH_PHOTO_STATE = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = this->unk_18E58;
+            } else {
+                gTrnsnUnkState = 2;
+                this->pauseBgPreRender.fbufSave = (u16*)gfxCtx->zbuffer;
+                this->pauseBgPreRender.cvgSave = NULL;
+            }
```

Comment:
they are very similar yeah. main difference is that `PAUSE_BG_PRERENDER_DONE` for the pause menu has more of a drastic effect on the game, in that it will allow the pause menu and other "overlay elements" as oot currently calls them to draw after the screenshot/prerender has already been taken.

The "setup" step for both avoids the call to the "overlay elements" or "post world elements" as both games call it currently, so that the hud and other elements arent shown in the picto picture or pause background. 
Process for both processes the image with prerender filters via `PreRender_ApplyFilters`, and the picto has the extra conversion step as well.

It seems like its prob a good idea that they line up. But I would take "bg" out of your current suggestion. its only called bg for pause because its in the background of the pause menu. its not a background for the picto picture

Reply:
Hmm, looking at it, even the Motion Blur status mirrors it as well
```c
typedef enum {
    /* 0 */ MOTION_BLUR_STOPPED,
    /* 1 */ MOTION_BLUR_INITALIZING,
    /* 2 */ MOTION_BLUR_RUNNING
} MotionBlurStatus;
```
Kind of seems like how play processes multistep process systems.

Reply:
Except possibly `gTrnsnUnkState` which seems close to this pattern but not quite.

Reply:
So I've synced all three enums, but haven't combined any yet. I'm not opposed to it, but I'm kind of intersted in which should be combined, and would like to know more about `gTrnsnUnkState` as well. Other's opinions would also be appreciated.

Reply:
atm I think the blur one is not as closely related to the picto and pause menu prerender states. they are similar in that they have "inactive, init, running", whatever terms you want to use for them. but the picto and pause menu ones are very close in purpose in terms of directing control flow of gameplay draw as it takes the screenshots and processes it etc

Reply:
True, I forgot about the control flow in draw. So it makes more sense that those need to actually stay in sync. Will combine those into one enum `PictoPreRenderState`.

Reply:
im not totally sure on combining enums. and at least with that naming scheme, it makes it sound like its only about the picto prerender and not the pause prerender.
imo currently its safe to say that they should be synced up in naming scheme, but can probably be different enums. but thats just imo